framework/src/modules/chain/loader.js

```diff
@@ -338,118 +329,116 @@ __private.getTransactionsFromNetwork = async function() {
  * Loads blocks from network.
  *
  * @private
- * @param {function} cb
- * @returns {setImmediateCallback} cb, err
+ * @returns {Promise} void
  * @todo Add description for the params
  */
-__private.loadBlocksFromNetwork = function(cb) {
+__private.loadBlocksFromNetwork = async () => {
 	// Number of failed attempts to load from the network.
 	let failedAttemptsToLoad = 0;
 	// If True, own node's db contains all the blocks from the last block request.
 	let loaded = false;
+	while (!loaded && failedAttemptsToLoad < 5) {
+		// eslint-disable-next-line no-loop-func
+		const loadingSteps = async () => {
```

Comment:
maybe better to remove this internal function definition and just call step by step is enough
It looks more complicated than it should be